imkar/scattering/freefield.py

```diff
@@ -0,0 +1,109 @@
+"""Scattering calculation functions based on free-field data."""
+import numpy as np
+import pyfar as pf
+
+
+def correlation_method(
+        sample_pressure, reference_pressure, microphone_weights):
+    r"""
+    Calculate the scattering coefficient from free-field reflection pattern.
+
+    This function uses the Mommertz correlation method [#]_ to compute the
+    scattering coefficient by averaging over the microphone positions:
+
+    .. math::
+        s = 1 -
+            \frac{|\sum_{\Omega_r} \underline{p}_{\text{sample}}(\Omega_r)
+            \cdot \underline{p}_{\text{reference}}^*(\Omega_r)
+            \cdot w(\Omega_r)|^2}
+            {\sum_{\Omega_r} |\underline{p}_{\text{sample}}(\Omega_r)|^2
+            \cdot w(\Omega_r) \cdot \sum_{\Omega_r}
+            |\underline{p}_{\text{reference}}(\Omega_r)|^2
+            \cdot w(\Omega_r) }
+
+    where:
+        - :math:`\underline{p}_{\text{sample}}` is the reflected sound
+          pressure of the sample under investigation.
+        - :math:`\underline{p}_{\text{reference}}` is the reflected sound
+          pressure from the reference sample.
+        - :math:`\Omega_r` represents the solid angle of the microphone
+          positions and :math:`w(\Omega_r)` represents its area weights.
+
+    Parameters
+    ----------
+    sample_pressure : pyfar.FrequencyData, pyfar.Signal
+        Reflected sound pressure or directivity of the test sample. Its cshape
+        must be (..., microphone_weights.size) and broadcastable to the
+        cshape of ``reference_pressure``. The frequency vectors of both
+        ``sample_pressure`` and ``reference_pressure`` must match.
+    reference_pressure : pyfar.FrequencyData, pyfar.Signal
+        Reflected sound pressure or directivity of the reference sample. Its
```

Comment:
What is meant with the "or"? Do reflected sound pressure and directivity mean the same thing or can both be used as input for the function?

Reply:
both can be used, because it is proportional